Resources/Post-Processors/Fusion 360/g2core 4th Axis Post.cps

```diff
@@ -0,0 +1,839 @@
+  "Invalid coolant mode"
+);
+
+var nFormat = createFormat({prefix:"N", decimals:0});
+var gFormat = createFormat({prefix:"G", decimals:0});
+var mFormat = createFormat({prefix:"M", decimals:0});
+var hFormat = createFormat({prefix:"H", decimals:0}); // not supported at the moment
+var dFormat = createFormat({prefix:"D", decimals:0}); // not supported at the moment
+
+var xyzFormat = createFormat({decimals:(unit == MM ? 3 : 4), forceDecimal:true});
+var rFormat = xyzFormat; // radius
+var abcFormat = createFormat({decimals:3, forceDecimal:true, scale:DEG});
+var feedFormat = createFormat({decimals:(unit == MM ? 0 : 1), forceDecimal:true});
+var toolFormat = createFormat({decimals:0});
+var rpmFormat = createFormat({decimals:0});
+var secFormat = createFormat({decimals:3, forceDecimal:true}); // seconds - range 0.001-99999.999
+var milliFormat = createFormat({decimals:0}); // milliseconds // range 1-9999
+var taperFormat = createFormat({decimals:1, scale:DEG});
+
+var xOutput = createVariable({prefix:"X"}, xyzFormat);
+var yOutput = createVariable({prefix:"Y"}, xyzFormat);
+var zOutput = createVariable({prefix:"Z"}, xyzFormat);
+var aOutput = createVariable({prefix:"A"}, abcFormat);
+var bOutput = createVariable({prefix:"B"}, abcFormat);
+var cOutput = createVariable({prefix:"C"}, abcFormat);
+var feedOutput = createVariable({prefix:"F"}, feedFormat);
+var sOutput = createVariable({prefix:"S", force:true}, rpmFormat);
+var dOutput = createVariable({}, dFormat);
+
+// circular output
+var iOutput = createReferenceVariable({prefix:"I", force:true}, xyzFormat);
+var jOutput = createReferenceVariable({prefix:"J", force:true}, xyzFormat);
+var kOutput = createReferenceVariable({prefix:"K", force:true}, xyzFormat);
+
+var gMotionModal = createModal({}, gFormat); // modal group 1 // G0-G3, ...
+var gPlaneModal = createModal({onchange:function () {gMotionModal.reset();}}, gFormat); // modal group 2 // G17-19
+var gAbsIncModal = createModal({}, gFormat); // modal group 3 // G90-91
+var gFeedModeModal = createModal({}, gFormat); // modal group 5 // G93-94
+var gUnitModal = createModal({}, gFormat); // modal group 6 // G20-21
+var gRetractModal = createModal({}, gFormat); // modal group 10 // G98-99
+
+var WARNING_WORK_OFFSET = 0;
+
+// collected state
+var sequenceNumber;
+var currentWorkOffset;
+
+/**
+  Writes the specified block.
+*/
+function writeBlock() {
+  if (properties.showSequenceNumbers) {
+    writeWords2(nFormat.format(sequenceNumber % 100000), arguments);
+    sequenceNumber += properties.sequenceNumberIncrement;
+  } else {
+    writeWords(arguments);
+  }
+}
+
+/**
+  Output a comment.
+*/
+function writeComment(text) {
+  writeln("(" + filterText(String(text).toUpperCase(), permittedCommentChars) + ")");
+}
+
+function onOpen() {
+	
+	if (true) {
+    var aAxis = createAxis({coordinate:0, table:true, axis:[-1, 0, 0], cyclic:true, preference:1});
+    machineConfiguration = new MachineConfiguration(aAxis);
```

Comment:
As a future enhancement, it's probably be useful to have the 4th axis be configurable, as some people have theirs as a B axis (rotates around Y) rather than as an A axis (rotates around X). :smile:

Reply:
Although technically incorrect nothing would prevent a user from labeling their B_AXIS A in g2, and then in fusion 360 defining the work coordinate setup as Y on axis with rotation and at 90.  the post would then still work,
I can't seem to get it to do that I've added an option that allows the user to turn off the A_AXIS code output however rotary and other 4th and 3+1 toolpaths would still output but without any of the a_axis code